### ISSUE TITLE
[Bug] Using InternVL-14B-224px to  process cross-modal retrieval but the logits result is not correct

### ISSUE THREAD

**Issue by Kenneth-X** [OpenGVLab/InternVL#759]
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.
- [X] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.

### Describe the bug

- **using InternVL-C(ontrastive) and InternVL-G(enerative) for cross-modal retrieval**
<img width="1552" alt="image" src="https://github.com/user-attachments/assets/f46d9f00-6f79-4a5d-a8c9-0bb991c415e6" />

- **when i delete the line 【prefix = 'summarize:'】it goes wrong ,  see the screenshot below**
<img width="953" alt="image" src="https://github.com/user-attachments/assets/25c8a910-e39f-43aa-8aa2-9cd40898fb6c" />

- **Is the prefix necessary? Can it be replace to other words like 【describe: 】？**

### Reproduction

import torch
from PIL import Image
from transformers import AutoModel, CLIPImageProcessor
from transformers import AutoTokenizer

model = AutoModel.from_pretrained(
    'model/InternVL-14B-224px',
    torch_dtype=torch.bfloat16,
    low_cpu_mem_usage=True,
    trust_remote_code=True).cuda().eval()

image_processor = CLIPImageProcessor.from_pretrained('model/InternVL-14B-224px')

tokenizer = AutoTokenizer.from_pretrained(
    'model/InternVL-14B-224px', use_fast=False, add_eos_token=True)
tokenizer.pad_token_id = 0  # set pad_token_id to 0
images = [
    Image.open('./examples/image1.jpg').convert('RGB'),
    Image.open('./examples/image2.jpg').convert('RGB'),
    Image.open('./examples/image3.jpg').convert('RGB')
]
prefix = ''
texts = [
    prefix + 'a photo of a red panda',  # English
    prefix + '一张熊猫的照片',  # Chinese
    prefix + '二匹の猫の写真'  # Japanese
]
pixel_values = image_processor(images=images, return_tensors='pt').pixel_values
pixel_values = pixel_values.to(torch.bfloat16).cuda()
input_ids = tokenizer(texts, return_tensors='pt', max_length=80,
                      truncation=True, padding='max_length').input_ids.cuda()

logits_per_image, logits_per_text = model(
    image=pixel_values, text=input_ids, mode='InternVL-C')
probs = logits_per_image.softmax(dim=-1)
print('\n')
print(probs)

### Environment

```Shell
pytorch 2.0
cuda 11.8
```

**Comment by czczup** [comment]
Hello, thank you for your attention. The prefix `prefix='summarize:'` is mandatory and cannot be changed. Initially, we used this prefix during training because it provided some performance improvements. Later, we discovered that it could not be removed or altered. Therefore, during inference, this prefix must be used to ensure consistency with the training phase.

**Comment by Kenneth-X** [comment]
> Hello, thank you for your attention. The prefix `prefix='summarize:'` is mandatory and cannot be changed. Initially, we used this prefix during training because it provided some performance improvements. Later, we discovered that it could not be removed or altered. Therefore, during inference, this prefix must be used to ensure consistency with the training phase.

thank you ,helps me a lot
